### ISSUE TITLE
Improve discoverability of available/installable output renderers for given mime types

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#131031]
## Summary
This issue captures the design explorations for improving the discoverability of output renderers. There are two main scenarios we want to focus on:

- When a notebook tries to use a renderer that is unavailable
- When a notebook uses a renderer that the user did not expect to see

Given these two main scenarios, we want to explore ways to increase the visibility of output renderers.

## Display badges

One simple idea would be to show a "badge" when there is an output renderer unavailable and also show an option to search the marketplace for other renderers:

![image](https://user-images.githubusercontent.com/35271042/129783420-b6c7ed24-212f-4549-9c18-d6fdd616bf78.png)

<img width="735" alt="CleanShot 2021-08-17 at 11 48 58@2x" src="https://user-images.githubusercontent.com/35271042/129783519-e517f95f-1f58-4cab-b5a1-44a857edc0a1.png">

## Status bar item

We could also add a status bar item for more of a seamless plug that would point to the same quick pick from above:

<img width="760" alt="CleanShot 2021-08-17 at 11 49 21@2x" src="https://user-images.githubusercontent.com/35271042/129783659-bba1ffb2-de8f-47b2-963b-bf4bfbef8568.png">

In the case where we have multiple outputs, we could consolidate the status bar items and show a list of the outputs before (though it's not best experience correlating the output list + visuals):

<img width="752" alt="CleanShot 2021-08-17 at 11 54 17@2x" src="https://user-images.githubusercontent.com/35271042/129784222-658fadd1-34ab-49ac-8664-b50d2c52a64f.png">

<img width="650" alt="CleanShot 2021-08-17 at 11 53 22@2x" src="https://user-images.githubusercontent.com/35271042/129784161-eb0b45e1-cb00-4f70-a2c5-c4eed27b1f0c.png">

And a "left-aligned" alternative:

<img width="804" alt="CleanShot 2021-08-17 at 11 54 59@2x" src="https://user-images.githubusercontent.com/35271042/129784357-b4812368-aee7-4678-bd99-1c1faba01aae.png">

## New output toolbar

We could move the output actions (currently in the ellipsis) into a new toolbar above each output, which would also be a place to highlight missing renderers:

<img width="766" alt="CleanShot 2021-08-17 at 11 59 33@2x" src="https://user-images.githubusercontent.com/35271042/129785338-8510678a-939c-4fe5-8cca-6dff1b58c5c7.png">

An alternative would be to place alignment on the right and use a dropdown for quick changes. However, there is limited space in the dropdown to communicate and would contradict other patterns where we use the quick pick:

<img width="807" alt="CleanShot 2021-08-17 at 12 07 04@2x" src="https://user-images.githubusercontent.com/35271042/129785814-046003e6-9db2-4807-aa1d-cab3675f1f45.png">

**Comment by misolori** [comment]
## Proposal

Based on the constraints of needing to account for multiple renderers, my proposal is to show a warning icon under each output where applicable for unavailable renderers. And we can add a new toolbar option for the output where they are inline and above the outputs (if we want to make it louder). 

![CleanShot 2021-08-17 at 16 23 58](https://user-images.githubusercontent.com/35271042/129813687-02781229-88e8-45e5-a5ef-2a39bfecfdab.gif)

**Comment by IanMatthewHuff** [comment]
The warning icon in the toolbar that you have seems smart, the initial "..." plus blue badge didn't seem as clear as the warning icon. 

If we can put the warning icon below the "..." in the vertical output toolbar could we unstack the other icons (select output and clear outputs) from the "..."? Right now always having an overflow "..." and only that looks kinda odd. I like the horizontal toolbar as I could actually click clear outputs without clicking an overflow and then a second click.

**Comment by misolori** [comment]
@IanMatthewHuff we can't unstack those icons as we need a solution that works for single line outputs:

<img width="217" alt="CleanShot 2021-08-18 at 07 12 09@2x" src="https://user-images.githubusercontent.com/35271042/129913895-8048bcf6-7166-49ae-9cf8-5c33a576e992.png">

So I think the option to have the output toolbar visible would probably help your use case.

**Comment by IanMatthewHuff** [comment]
@misolori Yeah, that makes sense, but then in the single line case, where does the missing renderer warning go? Since it looks stacked below the overflow icon?

**Comment by misolori** [comment]
@IanMatthewHuff that's a good point, it's probably more rare for this to happen since it would likely happen on cells with charts, but we could extend the cell to make room for the warning icon. Also, the output `...` overflow icon allows for that menu to grow/scale over time as well.
